src/components/focus_trap/focus_trap.tsx

```diff
@@ -72,14 +80,31 @@ export class EuiFocusTrap extends Component<EuiFocusTrapProps, State> {
     node.setAttribute('data-autofocus', 'true');
   };
 
+  onClickOutside = (e: MouseEvent | TouchEvent) => {
+    this.removeCloseListener();
+    // Timeout gives precedence to the consumer to initiate close if it has toggle behavior.
+    // Otherwise this event may occur first and the consumer toggle will reopen the flyout.
+    setTimeout(() => this.props.onClickOutside && this.props.onClickOutside(e));
+  };
+
+  addCloseListener = () => {
+    document.addEventListener('mouseup', this.onClickOutside);
+    document.addEventListener('touchend', this.onClickOutside);
+  };
+
+  removeCloseListener = () => {
```

Comment:
Similarly, I think this makes more sense to rename to be more specific to `mouseup`

```suggestion
  addMouseupListener = () => {
    document.addEventListener('mouseup', this.onClickOutside);
    document.addEventListener('touchend', this.onClickOutside);
  };

  removeMouseupListener = () => {
```

WYDT?